Feature/oee user guide (#39)
TamakiControl/tamaki-mes-docs: docs/user-guides/oee-guide/setup.md

---
sidebar_position: 2
title: "Setup"
description: "OEE Setup Configuration"
---

# Setup

**Navigation:**

![Navigation menu showing the Setup selection in the OEE module](./images/setup-navigation.png)

**Example:**

> ![Complete Setup screen showing configuration tabs and location tree](./images/setup.png)

Reviewer: Get a better image with more realistic location model. Basically just remove all but the TamakiControl location and then show its expanded tree in this image.

### OEE Configuration Fields

#### Export

Exports the OEE configuration for the selected location to an Excel file. This can be useful for backup purposes or for transferring configurations between different instances of Tamaki MES.

#### Delete

Deletes the OEE configuration for the selected location. This will not affect any existing OEE Records at this location, they will remain.

#### Enabled

Toggles the OEE tracking for the selected location.

#### Max Record Duration

Maximum duration for an OEE record in seconds. Normally, the following events cause an OEE record to end:
- OEE Mode change at the location.
- Standard rate change at the location.
- Production order change at the location.
- [future] Operation start or end at the location.
- [future] Shift start or end at the location.

If none of those events occur within the [Max Record Duration](#max-record-duration), the OEE record will end automatically. The purpose of this is to control the granularity of the records. Changing this value will not affect existing records, but will apply to new records created after the change.

#### Prune Days

> Number of days to retain OEE records. (If blank, records will be kept indefinitely)

Reviewer: ```suggestion
Number of days to retain OEE records in the database before deleting. (If blank, records will be kept indefinitely)
```

#### [Mode](terms-and-definitions#mode) Expression

The [Expression Field](terms-and-definitions#expression-field) is evaluated to an integer value corresponding to a [Mode](terms-and-definitions#mode)'s code of the location is bound to the OEE model.

#### Availability

##### Enable Availability

Enables availability tracking for the OEE Records. This allows the system to track when the location is running and when it is not, providing insights into the operational efficiency of the equipment.

##### State Expression

> The [Expression Field](terms-and-definitions#expression-field) where the PLC tag for the [State](terms-and-definitions#state) of the location is bound to the OEE model.

Reviewer: ```suggestion
The where the Oee State's code number is specified for this location [State](terms-and-definitions#state). This can be a static value but will most likely be a tag binding to reflect the PLC's value for the state of the machine [Expression Field](terms-and-definitions#expression-field)
```

##### Production Count Unit of Measure

The unit of measure for the [Production Count](terms-and-definitions#production-count). This is used to define how the production count is measured, such as in pieces, kilograms, or liters.

##### Production Rate Time Unit

The time unit for the production rate, such as seconds, minutes, or hours. This is used to define how the production rate is calculated. 

For example, if this field is set to `Minutes`, and the [Production Count Unit of Measure](#production-count-unit-of-measure) is set to `Pound`, then the production rate will be calculated in `Pounds per Minute`.

##### Production Count Expression

The [Expression Field](terms-and-definitions#expression-field) where the PLC tag for the [Production Count](terms-and-definitions#production-count) of the location is bound to the OEE model.

##### Production Count Calculation Type

Opens a dropdown with the following options:
- **Direct:** The production count will be calculated directly from the value in the [Production Count Expression](#production-count-expression) field.

> - **Delta:** The production count will be calculated as the difference between the current value and the previous value. This uses the [Production Count Overflow Value](#production-count-overflow-value) to know when the value has rolled over back to 0.

Reviewer: ```suggestion
- **Delta:** The production count will be calculated as the difference between the current value and the previous value. This uses the [Production Count Overflow Value](#production-count-overflow-value) to know when the value has rolled over back to 0. This only handles a maximum of 1 rollover between evaluations.

##### Production Count Overflow Value
The tag capturing production count will roll over to 0 when it reaches this value. This is used to calculate the production count when then [Production Count Calculation Type](#production-count-calculation-type) is set to `delta`.

##### Standard Rate Source

Opens a dropdown with the following options:
- **Static:** The standard rate will be a fixed value set in the [Standard Rate](#standard-rate-uomtime-unit) field.
- **Expression:** The standard rate will be determined by the [Standard Rate Expression](#standard-rate-expression-uomtime-unit) field.

##### Standard Rate  ([UoM](#production-count-unit-of-measure)/[Time Unit](#production-rate-time-unit))

The fixed value for the [Standard Rate](terms-and-definitions#standard-rate) of the location, defined in the unit of measure and time unit specified in the [Production Count Unit of Measure](#production-count-unit-of-measure) and [Production Rate Time Unit](#production-rate-time-unit) fields. This is used to define the expected production rate for the location.

##### Standard Rate Expression ([UoM](#production-count-unit-of-measure)/[Time Unit](#production-rate-time-unit))

> The [Expression Field](terms-and-definitions#expression-field) where the PLC tag for the [Standard Rate](terms-and-definitions#standard-rate) of the location is bound to the OEE model.

Reviewer: ```suggestion
This is where the value for the [Standard Rate](terms-and-definitions#standard-rate) is specified for the location. This can be a static value but will most likely be a tag binding to reflect the PLC's value for the standard rate for the product running at the machine [Expression Field](terms-and-definitions#expression-field).
```

#### Quality

##### Enable Quality

Enables quality tracking for the OEE Records. This allows the system to track the waste count and rate of the location.

##### Waste Count Expression

The [Expression Field](terms-and-definitions#expression-field) where the PLC tag for the [Waste Count](terms-and-definitions#waste-count) of the location is bound to the OEE model.

##### Waste Count Calculation Type

Opens a dropdown with the following options:
- **Direct:** The waste count will be calculated directly from the value in the [Waste Count Expression](#waste-count-expression) field.

> - **Delta:**  The waste count will be calculated as the difference between the current value and the previous value. This uses the [Waste Count Overflow Value](#waste-count-overflow-value) to know when the value has rolled over back to 0.

Reviewer: ```suggestion
- **Delta:**  The waste count will be calculated as the difference between the current value and the previous value. This uses the [Waste Count Overflow Value](#waste-count-overflow-value) to know when the value has rolled over back to 0. This only handles a maximum of 1 rollover between evaluations.
```
